### PR TITLE
package/mosquitto: add mosquitto

### PR DESCRIPTION
**Pull request by mjhuff**
Closes [EXEC-159](https://opentrons.atlassian.net/browse/EXEC-159)

The buildroot analog to https://github.com/Opentrons/oe-core/pull/124

Adds the Mosquitto message broker for MQTT notifications.

[EXEC-159]: https://opentrons.atlassian.net/browse/EXEC-159?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package/mosquitto/mosquitto.service [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=Mosquitto MQTT broker
+
+[Service]
+ExecStart=/usr/sbin/mosquitto -c /etc/mosquitto/mosquitto.conf
+ExecReload=/bin/kill -HUP $MAINPID
+Restart=always
+
+[Install]
+WantedBy=multi-user.target
```
Comment (SyntaxColoring) [inline package/mosquitto/mosquitto.service:10]: Do we need anything special to ensure it's started before (or after?) opentrons-robot-server.service?

Comment (mjhuff) [reply package/mosquitto/mosquitto.service:10]: Good catch 👍

File: package/mosquitto/mosquitto.service
Comment (SyntaxColoring) [inline package/mosquitto/mosquitto.service:1]: Organizationally, we probably want to leave this `package/mosquitto/mosquitto.service` file alone and preserve it as the pristine upstream. To customize it so it's treated as a dependency of `opentrons-robot-server`, I see a couple of options:

1. Add a [drop-in](https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html#:~:text=Along%20with%20a%20unit%20file%20foo.service%2C%20a%20%22drop%2Din%22%20directory%20foo.service.d/%20may%20exist) for `mosquitto.service`. [Here's an example of us doing that in `/usr/lib`](https://github.com/Opentrons/buildroot/blob/2060b716b7804c9a63a0038d795fd742a92c5685/board/opentrons/ot2/rootfs-overlay/usr/lib/systemd/system/avahi-daemon.service.d/run-late.conf), and [here's an example of us doing that in `/etc`](https://github.com/Opentrons/buildroot/blob/2060b716b7804c9a63a0038d795fd742a92c5685/board/opentrons/ot2/rootfs-overlay/etc/systemd/system/dropbear.service.d/create-host-key-directory.conf). I'm not sure if we have a preference for `/usr/lib` or `/etc` in general.
2. Leave `mosquitto.service` alone and modify `opentrons-robot-server.service` instead. [This is what we do for nginx](https://github.com/Opentrons/opentrons/blob/c3f1bf03386ac5eb6dc3ba231912aea962e70fd6/robot-server/opentrons-robot-server.service#L5-L6).

I would probably do 2, for consistency.

Comment (mjhuff) [reply package/mosquitto/mosquitto.service:1]: Gotcha, I'll do this in a follow-up PR then.

Comment (SyntaxColoring) [inline package/mosquitto/mosquitto.service:1]: Sorry for the continued back-and-forth—

Is it correct for this file to be completely deleted? Don't we want it to match upstream buildroot?

Comment (mjhuff) [reply package/mosquitto/mosquitto.service:1]: Woops just saw this - I appreciate all the feedback! Still very new to this.

I assumed that we implicitly use upstream service files if they aren't present locally, but is that incorrect?

Comment (SyntaxColoring) [reply package/mosquitto/mosquitto.service:1]: I think a package theoretically could do that, but the Buildroot convention seems to be to keep them in-repo. e.g. here's [apache's](https://github.com/Opentrons/buildroot/blob/opentrons-develop/package/apache/apache.service) (which we don't use). OpenEmbedded might work differently.

In other words, the Buildroot repo (or a fork of the Buildroot repo) contains recipes and default configurations for every package that someone might want a system to have in `package/`, and then you do all your customization for your system in `board/` and `configs/`.

**Review by sfoster1 — APPROVED** [review]
Sweet

**Review by SyntaxColoring — APPROVED** [review]

**Pull request merged**